### PR TITLE
Update README: status, relation to other projects

### PR DESCRIPTION
**Pull request by znewman01**
So that nobody who sees this is *too* confused.

I left the TSA off for now, figure that would only complicate things more :smile:

Signed-off-by: Zack Newman <zjn@chainguard.dev>

### CODE REVIEW

**Review by cpanato — DISMISSED** [review]
looks cool

**Review by asraa — DISMISSED** [review]
Love the diagram so far! 

I'm a little interested to know where the CLI clients for Rekor/Fulcio go, to me those are like "cosign" CLI tools (and in the long tail should be separated from the sigstore/rekor packages)

File: README.md [hunk 1]
```diff
@@ -1,8 +1,45 @@
 # sigstore-go 
 
-This is a Go library to interact with [sigstore](https://sigstore.dev/).
+This is a Go client library for [Sigstore](https://sigstore.dev/).
 
-The project is under early development.
+The project is under early development and **you shouldn't depend on it** yet. In the long run, you should depend on sigstore-go instead of cosign or sigstore/sigstore *unless* you need support for OCI registries/containers.
```
Comment (lukehinds) [inline README.md]: I would fix this text up, as it can be read as; if you want OCI support use cosign as a library

File: README.md [hunk 2]
```diff
@@ -1,8 +1,45 @@
...
+  sigstore-go --> fulcio
+  sigstore-go --> s/s
+  rekor --> s/s
+  fulcio --> s/s
+  rekor --> protobuf-specs
+  fulcio --> protobuf-specs
+  clients --> rekor
+  clients --> fulcio
+  sigstore-go --> protobuf-specs
+  clients("sigstore-{ruby,js,java,python,rs}") --> protobuf-specs
```
Comment (lukehinds) [inline README.md:26]: cosign uses s/s for its crypto key operations.

File: README.md [hunk 3]
```diff
@@ -1,8 +1,45 @@
...
+```
+
+See [Sigstore in Golang](https://docs.google.com/document/d/1aZfk1TlzcuaO0uz76M9D26-gAvoZLn0oCAKvkbuhcPM/edit) for design rationale; you may need to join sigstore-dev@googlegroups.com for access. We have:
+
+* Infrastructure
+  * [fulcio](https://github.com/sigstore/fulcio): The CA for Sigstore. Contains the Fulcio server implementation along with a basic generated client library. 
+  * [rekor](https://github.com/sigstore/rekor): The artifact log for Sigstore. Contains the Rekor server implementation along with a basic generated client library.
+  * [protobuf-specs](https://github.com/sigstore/protobuf-specs): Definitions for common Sigstore data formats.
+* Golang
+  * [sigstore/sigstore](https://github.com/sigstore/sigstore): Common code, used in both the infrastructure and clients.
```
Comment (lukehinds) [inline README.md:39]: This is helpful, we should update s/s with the same definition.

File: README.md (hunk 2 above)
Comment (znewman01) [reply README.md:26]: Good point! However, I think I'd prefer to tell a half-truth and avoid that edge here for two reasons:

1. The graph is already a little crowded/confusing.
2. In theory, we could remove this dependency (e.g., by reexporting all relevant functionality from s/s in sigstore-go). This aligns with the goals of the project—for downstream Sigstore users in Golang to have only one direct dependency.

File: README.md (hunk 3 above)
Comment (znewman01) [reply README.md:39]: I can do that! https://github.com/sigstore/sigstore/pull/817

File: README.md (hunk 1 above)
Comment (znewman01) [reply README.md]: Made an attempt—how's that?

**Comment by znewman01** [comment]
> I'm a little interested to know where the CLI clients for Rekor/Fulcio go, to me those are like "cosign" CLI tools (and in the long tail should be separated from the sigstore/rekor packages)

I'm gonna punt on that one 🙂  I think there are a number of potential good answers.

**Comment by asraa** [comment]
> I'm gonna punt on that one slightly_smiling_face I think there are a number of potential good answers.

Hahha I agree! Approval without it is fine :)

**Review by cpanato — APPROVED** [review]

**Comment by cpanato** [comment]
thanks for doing this

**Review by asraa — APPROVED** [review]

**Pull request merged**